CoffeeEngine/src/CoffeeEngine/ParticleManager/ParticleManager.cpp

```diff
@@ -169,13 +214,18 @@ namespace Coffee
         {
             if (velocityOverLifeTimeSeparateAxes)
             {
-                newVelocity.x = CurveEditor::GetCurveValue(normalizedLife, speedOverLifeTimeX);
-                newVelocity.y = CurveEditor::GetCurveValue(normalizedLife, speedOverLifeTimeY);
-                newVelocity.z = CurveEditor::GetCurveValue(normalizedLife, speedOverLifeTimeZ);
+                newVelocity.x = CurveEditor::ScaleCurveValue(CurveEditor::GetCurveValue(normalizedLife, speedOverLifeTimeX), -1, 1);
+                
+                newVelocity.y = CurveEditor::ScaleCurveValue(CurveEditor::GetCurveValue(normalizedLife, speedOverLifeTimeY), -1, 1);
+                
+                newVelocity.y = CurveEditor::ScaleCurveValue(CurveEditor::GetCurveValue(normalizedLife, speedOverLifeTimeZ), -1, 1);
```

Comment:
Shouldn't this be `newVelocity.z` instead of `newVelocity.y`? I presume that is a typo but let me know 

```suggestion
                newVelocity.z = CurveEditor::ScaleCurveValue(CurveEditor::GetCurveValue(normalizedLife, speedOverLifeTimeZ), -1, 1);
```

Reply:
Nope, to maintain consistency in the axes for the user, they need to be swapped.
The way the plane is made, when rotated to create the billboard, instead of being XYZ, it becomes XZY.
Since everything is designed to be XYZ, I made it so that the user inputs it as XYZ, while internally it is handled as XZY.

If you want, I can change this, but there will be parts where the user will have to input XYZ and others where they will have to input XZY.

Reply:
Oh, you're right, I see it now. Anyway, I need to make the 'y' be calculated with speedOverLifeTimeZ, and the 'z' with speedOverLifeTimeY, just like the size. Because of the use of the plane mesh in the particles and the user's interpretation of the axes.

Reply:
Yeah, I think the operations are correct. In the testing I've done, I just changed the axis where is stored and it seems to work fine, so probably its just the typo hashs :)